cmake/OpenCVDetectVTK.cmake

```diff
@@ -1,34 +1,34 @@
-# VTK 9.0
 if(NOT VTK_FOUND)
-  find_package(VTK 9 QUIET NAMES vtk COMPONENTS
```

Comment:
Can we save/restore `VTK_DIR` variable instead?

Reply:
It would work, but I think the approach used in this PR better conveys the intent to select appropriate version of a library as opposed to finding any version of a library with required list of components. Besides, saving and restoring `VTK_DIR` looks like a workaround, whereas the current implementation simply avoids this problem.